src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/ThunkPool.cs

```diff
@@ -347,16 +342,15 @@ public static unsafe IntPtr GetNewThunksBlock()
             }
             else
             {
-                nextThunksBlock = RuntimeImports.RhAllocateThunksMapping();
+                int isOOM = 0;
+                nextThunksBlock = RuntimeImports.RhAllocateThunksMapping(&isOOM);
 
                 if (nextThunksBlock == IntPtr.Zero)
                 {
-                    // We either ran out of memory and can't do anymore mappings of the thunks templates sections,
-                    // or we are using the managed runtime services fallback, which doesn't provide the
-                    // file mapping feature (ex: older version of mrt100.dll, or no mrt100.dll at all).
-
-                    // The only option is for the caller to attempt and recycle unused thunks to be able to
-                    // find some free entries.
+                    if (isOOM == 0)
```

Comment:
Can we throw the exception immediately here instead of propagating it manually? The manual error propagation looks like a left-over from the .NET Native MRT/app split.